Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm

```diff
@@ -39,6 +39,43 @@
 #import <WebKit/WKWebViewPrivateForTesting.h>
 #import <wtf/BlockPtr.h>
 
+@interface PopoverNotificationListener : NSObject
+- (instancetype)initWithCallback:(Function<void(NSNotification *)>&&)callback;
+@end
+
+@implementation PopoverNotificationListener {
+    Function<void(NSNotification *)> _callback;
+}
+
+- (instancetype)initWithCallback:(Function<void(NSNotification *)>&&)callback
+{
+    if (!(self = [super init]))
+        return nil;
+
+    auto *notificationCenter = NSNotificationCenter.defaultCenter;
+    [notificationCenter addObserver:self selector:@selector(handleNotification:) name:NSPopoverWillShowNotification object:nil];
+    [notificationCenter addObserver:self selector:@selector(handleNotification:) name:NSPopoverDidShowNotification object:nil];
+    [notificationCenter addObserver:self selector:@selector(handleNotification:) name:NSPopoverWillCloseNotification object:nil];
+    [notificationCenter addObserver:self selector:@selector(handleNotification:) name:NSPopoverDidCloseNotification object:nil];
+    _callback = WTFMove(callback);
+    return self;
+}
+
+- (void)dealloc
+{
+    [super dealloc];
```

Comment:
Nit: `[super dealloc]` should be the last statement in `dealloc` as a general guideline.

(though it's technically safe in this case since we are not accessing any variables of the superclass)